builder/quoter.go

```diff
@@ -0,0 +1,42 @@
+package builder
+
+import (
+	"strings"
+)
+
+// Quoter returns safe and valid SQL strings to use when building a SQL text.
+type Quoter interface {
+	// ID quotes identifiers such as schema, table, or column names.
+	// ID does not operate on multipart identifiers such as "public.Table",
+	// it only operates on single identifiers such as "public" and "Table".
+	ID(name string) string
+
+	// Value quotes database values such as string or []byte types as strings
+	// that are suitable and safe to embed in SQL text. The returned value
+	// of a string will include all surrounding quotes.
+	//
+	// If a value type is not supported it must panic.
+	Value(v interface{}) string
+}
+
+// SqlQuoter is default implementation of Quoter interface.
+type SqlQuoter struct {
+	IDPrefix             string
+	IDSuffix             string
+	IDSuffixEscapeChar   string
+	ValueQuote           string
+	ValueQuoteEscapeChar string
+}
+
+func (q *SqlQuoter) ID(name string) string {
+	return q.IDPrefix + strings.Replace(name, q.IDSuffix, q.IDSuffixEscapeChar+q.IDSuffix, -1) + q.IDSuffix
+}
+
+func (q *SqlQuoter) Value(v interface{}) string {
```

Comment:
no need to have pointer receiver for this struct?

Reply:
this is a small struct, should be fine I think, maybe just to be safe, can you try creating a small benchmark to choose pointer or not pointer? (I would like to avoid heap allocation as pointer is one of the possible cause)

aside from that, we can still remove pointer from the receiver because this method doesn't update any of the struct variable, and we may create this struct as pointer if it's shown to be better

Reply:
They both have pretty much the same performance, so I will change it to struct, adapter can still provide than both struct or pointer as Quoter if they need it.

```
BenchmarkBuffer_InlineValueStruct-8       100000               239.7 ns/op            48 B/op          4 allocs/op
BenchmarkBuffer_InlineValuePointer-8      100000               231.2 ns/op            48 B/op          4 allocs/op
```